### Die 148 grössten Städte der Schweiz

***
&Ouml;ffne cities.txt, lies alle Zeilen ein und entferne white-space von rechts
***

In [1]:
lines = []
# Filename anpassen, so dass dein File geoeffnet wird
with open('cities_.txt', mode = 'r') as f:
    for line in f:
        lines.append(line.rstrip())
        
lines[:4]        

[' \tSTADT \tKANTON \tEINWOHNER',
 "1. \tZürich \tZH \t396'027",
 "2. \tGenf \tGE \t194'565",
 "3. \tBasel \tBS \t168'620"]

In [13]:
lines[-2:] 

["147. \tZollikofen \tBE \t10'097", "148. \tRisch \tZG \t10'036"]

***
Trenne Zeilen beim Tabulator '\t', entferne Spaces von rechts,
verwandle rechte Spalte in Integers. Ersetzte vorg&auml;ngig
`"'"` duch `''` mit `str.replace("'",'')`.
***

In [2]:
table = []
for row in lines:
    cols = row.split('\t')
    cols = [c.rstrip() for c in cols]
    cols[-1] = cols[-1].replace("'",'')
    if cols[-1].isnumeric():
        cols[-1] = int(cols[-1])
    table.append(cols[1:])

In [3]:
table[:3]

[['STADT', 'KANTON', 'EINWOHNER'],
 ['Zürich', 'ZH', 396027],
 ['Genf', 'GE', 194565]]

***
Als Tabelle mit `table2str`
***

In [5]:
from toolbox import table2str
s= table2str(table[0], table[1:])
print(s)

 │ STADT                  │ KANTON │ EINWOHNER │ 
 │ ━━━━━━━━━━━━━━━━━━━━━━ │ ━━━━━━ │ ━━━━━━━━━ │ 
 │ Zürich                 │ ZH     │    396027 │ 
 │ Genf                   │ GE     │    194565 │ 
 │ Basel                  │ BS     │    168620 │ 
 │ Bern                   │ BE     │    140634 │ 
 │ Lausanne               │ VD     │    135629 │ 
 │ Winterthur             │ ZH     │    106778 │ 
 │ Luzern                 │ LU     │     81284 │ 
 │ St. Gallen             │ SG     │     75310 │ 
 │ Lugano                 │ TI     │     63668 │ 
 │ Biel/Bienne            │ BE     │     53667 │ 
 │ Thun                   │ BE     │     43303 │ 
 │ Köniz                  │ BE     │     39998 │ 
 │ La Chaux-de-Fonds      │ NE     │     39027 │ 
 │ Freiburg               │ FR     │     38288 │ 
 │ Schaffhausen           │ SH     │     35927 │ 
 │ Chur                   │ GR     │     34547 │ 
 │ Vernier                │ GE     │     34477 │ 
 │ Neuenburg              │ NE     │     33641 │ 


***
Dictionary erstellen mit St&auml;dtenamen als Keys.  
Die Werte sind Dictionaries der Form  
`{'KANTON': 'ZH', 'EINWOHNER': 396027}`
***

In [8]:
cities = {}
keys =table[0][1:]
for row in table[1:]:
    cities[row[0]] = dict(zip(keys,row[1:]))

In [12]:
{k:cities[k] for k in list(cities)[:3]}

{'Zürich': {'KANTON': 'ZH', 'EINWOHNER': 396027},
 'Genf': {'KANTON': 'GE', 'EINWOHNER': 194565},
 'Basel': {'KANTON': 'BS', 'EINWOHNER': 168620}}